trin-core/src/jsonrpc/types.rs

```diff
@@ -70,6 +77,38 @@ fn validate_jsonrpc_version(jsonrpc: &str) -> Result<(), ValidationError> {
     Ok(())
 }
 
+pub struct PingParams {
+    pub enr: Enr,
+}
+
+impl TryFrom<Params> for PingParams {
+    type Error = ValidationError;
+
+    fn try_from(params: Params) -> Result<Self, Self::Error> {
+        match params {
+            Params::Array(val) => match val.len() {
+                1 => PingParams::try_from(&val[0]),
```

Comment:
Yeah, this seems like a solid solution to not wanting to nest too many `match`es.